### PR TITLE
Fix the definition of the `PDH_COUNTER_INFO` struct

### PR DESCRIPTION
**Pull request by SeeminglyScience**
# PR Summary

Fixes the crash from #13829

## PR Context

The definition is mostly generated via [dotnet/ClangSharp](https://github.com/dotnet/ClangSharp) and altered a little manually. It's a bit involved for a struct we only use two fields from but it's easy to include in full so if we end up needing more this makes it very easy.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
@SeeminglyScience please fix the CodeFactor issue if they were introduced in this PR: https://www.codefactor.io/repository/github/powershell/powershell/pull/17779

**Comment by adityapatwardhan** [comment]
@SeeminglyScience can you validate the change on x86 and arm64? We do not run CI on those platforms.

**Comment by SeeminglyScience** [comment]
> @SeeminglyScience can you validate the change on x86 and arm64? We do not run CI on those platforms.

Tested in x86, looks good. Arm64 I don't have the ability to test afaik

**Comment by SeeminglyScience** [comment]
@adityapatwardhan Verified with Steve that `Arm64` works as well 🎉

File: src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs [hunk 1]
```diff
@@ -267,28 +267,65 @@ private struct PDH_TIME_INFO
         // We only need dwType and lDefaultScale fields from this structure.
         // We access those fields directly. The struct is here for reference only.
         //
-        [StructLayout(LayoutKind.Explicit, CharSet = CharSet.Unicode)]
-        private struct PDH_COUNTER_INFO
+        [StructLayout(LayoutKind.Sequential)]
```
Comment (adityapatwardhan) [inline src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs:270]: Do we need CharSet?

Comment (SeeminglyScience) [reply src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs:270]: I can add it if preferred but the struct is directly blittable so it's not strictly necessary.

**Review by adityapatwardhan — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs (hunk 1 above)
Comment (adityapatwardhan) [reply src/Microsoft.PowerShell.Commands.Diagnostics/PdhHelper.cs:270]: If not mandatory, we should not have it.

**Pull request merged**